spec/lib/whitehall_importer/import_spec.rb

```diff
@@ -57,6 +57,32 @@
       expect { described_class.call(document_import) }.not_to(change { User.count })
     end
 
+    it "does not create a user who has a nil uid" do
+      user = build(:whitehall_export_user, uid: nil)
+      import_data = build(:whitehall_export_document, users: [user])
+      document_import = build(:whitehall_migration_document_import, payload: import_data)
+
+      expect { described_class.call(document_import) }.not_to(change { User.count })
+      expect(Document.last.created_by_id).to be_nil
+    end
```

Comment:
This test looks good, I don't think we need the 2 subsequent tests though as they don't seem to be testing scenarios that aren't tested implicitly.

Do we know if Whitehall returns these uid values as nil, empty strings or both? We should be covered for both by the present method but if we care about the empty string we may want to test on that.

Reply:
Not sure how Whitehall returns these - will take a look and create any additional tests and document.

Reply:
Have updated this test to also validate that the user count does not increase.